### ISSUE TITLE
Hyper parameter tuning set timeouts

### ISSUE THREAD

**Issue by gsagarvarma** [johannespetrat/HyperparameterOptimisation#12]
It would be useful to add a way to stop the hyper-parameter tuning early by some condition like
1. The loss has fallen below a certain threshold
2. Time taken exceeds the limits
3. Limiting the number of iterations/combinations
 - instead of always continuing the search until maximum evaluations is reached.
